rwengine/src/script/ScriptTypes.cpp

```diff
@@ -95,3 +95,144 @@ template<> GameObject* ScriptArguments::getObject< VehicleObject >(unsigned int
 	auto it = getWorld()->objects.find( gameObjectID );
 	return (it == getWorld()->objects.end())? nullptr : it->second;
 }*/
+
+template<> float ScriptArguments::getParameter<float>(unsigned int arg) const
+{
+	return 0.f;
+}
+
+template<> int ScriptArguments::getParameter<int>(unsigned int arg) const
+{
+	return getParameters()[arg].integerValue();
+}
+
+template<> const char* ScriptArguments::getParameter<const char*>(unsigned int arg) const
+{
+	return "";
+}
+
+template<> glm::vec2 ScriptArguments::getParameter<glm::vec2>(unsigned int arg) const
+{
+	RW_CHECK(getParameters().size() > arg+1, "Argument list too short to get vec2");
+	return glm::vec2(getParameter<float>(arg), getParameter<float>(arg+1));
+}
+
+template<> glm::vec3 ScriptArguments::getParameter<glm::vec3>(unsigned int arg) const
+{
+	RW_CHECK(getParameters().size() > arg+2, "Argument list too short to get vec3");
+	return glm::vec3(getParameter<float>(arg), getParameter<float>(arg+1), getParameter<float>(arg+2));
+}
+
+template<> glm::u8vec3 ScriptArguments::getParameter<glm::u8vec3>(unsigned int arg) const
+{
+	RW_CHECK(getParameters().size() > arg+2, "Argument list too short to get u8vec3");
+	return glm::u8vec3(getParameter<int>(arg), getParameter<int>(arg+1), getParameter<int>(arg+2));
+}
+
+template<> glm::u8vec4 ScriptArguments::getParameter<glm::u8vec4>(unsigned int arg) const
+{
+	RW_CHECK(getParameters().size() > arg+3, "Argument list too short to get u8vec4");
+	return glm::u8vec4(getParameter<int>(arg), getParameter<int>(arg+1), getParameter<int>(arg+2), getParameter<int>(arg+3));
+}
+
+template<> float& ScriptArguments::getParameterRef<float>(unsigned int arg) const
+{
+	auto& param = (*this)[arg];
+	return *((float*)param.handleValue());
```

Comment:
a) `reinterpret_cast` please

b) This will be wrong if `sizeof int != sizeof float`

c) This may violate strict aliasing rules anyway, you should use memcpy to copy the int into a float. (It's optimized out, but avoids UB).